### PR TITLE
Update Cloud to accept log messages from V2X Hub (SPaT) - JSON

### PR DESCRIPTION
**Pull request by dan-du-car**
# PR Details
## Description
As an ODE administrator, I want to accept and decode log messages from V2X Hub and publish on Kafka topics available for cloud or database consumers. These logs from the V2X Hub may come to the ODE via batch upload logs or Kafka topics.

## Related Issue
https://github.com/usdot-jpo-ode/jpo-ode/issues/415

## Motivation and Context
ODE and V2xHub integration

## How Has This Been Tested?
Local integration test

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[ODE Contributing Guide](https://github.com/usdot-jpo-ode/jpo-ode/blob/bugfix/Pull_request_template/docs/contributing_guide.md) 
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by dan-du-car** [comment]
Local integration testing result:
Producer:
producer

consumer:
consumer

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package us.dot.its.jpo.ode.model;
+
+public class OdeSpatMetadata extends OdeLogMetadata {
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -5361008186032548625L;
+
+	public enum SpatSource {
```
Comment (snallamothu) [inline jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java:9]: @dan-du-car The way they defined this is confusing.  Seems like this enumeration is being used to mark when the message is received or being sent.  Check with Tony or Brandon if this applies to Spat messages also.

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java [hunk 2]
```diff
@@ -47,74 +50,118 @@ public Asn1DecodeMessageJSON(OdeProperties odeProps) {
 	protected Object process(String consumedData) {
 		OdeData odeData = null;
 		OdeMsgPayload payload = null;
-		OdeBsmMetadata metadata = null;
 		try {
 			JSONObject rawJSONObject = new JSONObject(consumedData);
 			Set<?> keys = rawJSONObject.keySet();
 			for (Object key : keys) {
 				
 				//Send encoded BSM content to Codec service to decode BSM
```
Comment (snallamothu) [inline jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java:58]: There is lot of code duplication below, there is an opportunity for some refactoring.  OK, as if for now.

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java [hunk 3]
```diff
@@ -47,74 +50,118 @@ public Asn1DecodeMessageJSON(OdeProperties odeProps) {
...
 			logger.info("Sending encoded payload XML to ASN1 codec {}", xmlUtils.toXml(odeData));
-			codecPublisher.publish(xmlUtils.toXml(odeData),codecPublisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
+			codecPublisher.publish(xmlUtils.toXml(odeData),
+					codecPublisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
 		} catch (JsonProcessingException e) {
 			logger.info("Error sending encoded payload XML to ASN1 codec {}", e.getMessage());
 			e.printStackTrace();
 		}
-		
+
```
Comment (snallamothu) [inline jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java:165]: @dan-du-car Didn't see any unit tests.  Sonar scan is reporting 0% coverage.  We should add unit tests for at least the new code if none exist.

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeLogMetadata.java [hunk 4]
```diff
@@ -23,7 +23,7 @@
    private static final long serialVersionUID = -8601265839394150140L;
 
    public enum RecordType {
-      bsmLogDuringEvent, rxMsg, dnMsg, bsmTx, driverAlert, unsupported
+      bsmLogDuringEvent, rxMsg, dnMsg, bsmTx, driverAlert, unsupported, spatTx
```
[the remark this answers is not in the document]
Comment (TonyEnglish) [reply jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeLogMetadata.java:26]: Agreed, spatTx would make more sense before unsupported, and i am also not sure what unsupported would be used for

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java (hunk 1 above)
Comment (TonyEnglish) [reply jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java:9]: i think this enum would be for bsm source (EV/RV meaning local vehicle broadcast vs remove vehicle broadcast), for SPaT i think the source would be RSU, V2X Hub, and maybe MMITSS

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java (hunk 2 above)
Comment (TonyEnglish) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java:58]: Agreed, and work with Drew on the TIMs (so now we have 3 messages, with many more to come) to plan the standard approach.

**Review by TonyEnglish — COMMENTED** [review]
Few comments for consideration

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java (hunk 1 above)
Comment (dan-du-car) [reply jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java:9]: okay, update to RSU, V2X, MMITSS, unknown?

File: jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java (hunk 3 above)
Comment (dan-du-car) [reply jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java:165]: @snallamothu Currently all the unit test is set to be ignored. We have another existing PR addressing the unit tests.

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeLogMetadata.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (dan-du-car) [reply jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeLogMetadata.java]: updated

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java (hunk 1 above)
Comment (TonyEnglish) [reply jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java:9]: that make sense

**Review by TonyEnglish — APPROVED** [review]

File: jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java (hunk 1 above)
Comment (dan-du-car) [reply jpo-ode-core/src/main/java/us/dot/its/jpo/ode/model/OdeSpatMetadata.java:9]: updated https://github.com/usdot-jpo-ode/jpo-ode/pull/414/commits/8162dc2ad2d02e6ed502768f7c5a2ab8e1f31463

**Review by snallamothu — APPROVED** [review]
